tests/cli/test_cli_utils.py

```diff
@@ -64,3 +65,54 @@ def test_table_matches_patterns(self):
             pattern="telemetry_live.event_v4",
             invert=True,
         )
+
+    def test_extract_last_group_by_from_query(self):
+        assert ["ALL"] == extract_last_group_by_from_query(
+            query="SELECT column_1 FROM test_table GROUP BY ALL"
+        )
+        assert ["1"] == extract_last_group_by_from_query(
+            query="SELECT column_1, SUM(metric_1) AS metric_1 FROM test_table GROUP BY 1;"
```

Comment:
the tests don't pass if I lower case `group by` ... which should never happen according to our style guide .. but since you are testing on `GROUP BY 1` which should also never happen, I wanted to flag it 🎏

Reply:
TY! I updated the function to case insensitive.